contracts/staking/fuse/utils/StakingRewardsPerEpoch.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract StakingRewardsPerEpoch is AccessControl, ReentrancyGuard, Pausable {
+	using SafeERC20 for IERC20;
+
+	struct StakerInfo {
+		uint256 reward;
+		uint256 balance;
+		uint256 pendingStake;
+		uint256 indexOfLastEpochStaked;
+	}
+
+	event Staked(address indexed staker, uint256 amount);
+	event Withdrawn(address indexed staker, uint256 amount);
+	event RewardPaid(address indexed user, uint256 reward);
+	event Recovered(address token, uint256 amount);
+
+	bytes32 public constant GUARDIAN_ROLE = keccak256("GUARDIAN_ROLE");
+	uint256 public constant PRECISION = 1e18;
+
+	IERC20 public rewardsToken;
+	IERC20 public stakingToken;
+
+	mapping(address => StakerInfo) public stakersInfo;
+
+	uint256 public lastEpochIndex;
+	uint256 public pendingStakes;
+	uint256[] public rewardsPerTokenAt;
+
+	uint256 public totalSupply;
+
+	modifier updateReward(address account) {
+		_updateReward(account);
+		_;
+	}
+
+	constructor(address _rewardsToken, address _stakingToken) {
+		_setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+		_setupRole(GUARDIAN_ROLE, msg.sender);
+		rewardsToken = IERC20(_rewardsToken);
+		stakingToken = IERC20(_stakingToken);
+	}
```

Comment:
no need for that, this contract should not handle token transfers

Reply:
done removal